src/App.tsx

```diff
@@ -0,0 +1,30 @@
+import { ReactElement, useState } from 'react';
+
+import './App.css';
+import { Counter } from './Counter/Counter.tsx';
+import { Header } from './Header/Header.tsx';
+import { GenreSelect, SearchForm } from './components';
+
+function App(): ReactElement {
+	const [count, setCount] = useState(0);
+	const [searchTerm, setSearchTerm] = useState('');
+	const [selectedGenre, setGenre] = useState<string | null>('All');
+	const [genres] = useState(['All', 'Documentary', 'Comedy', 'Horror', 'Crime']);
+
+	return (
+		<>
+			<Header>
+				<SearchForm searchTerm={searchTerm} onSearch={(setSearchTerm)}></SearchForm>
+			</Header>
+			<main className={'card'}>
+				<GenreSelect selectedGenre={selectedGenre} genres={genres} onSelect={setGenre}></GenreSelect>
+				<Counter count={count}
+					onIncrease={() => setCount((prev: number): number => prev + 1)}
```

Comment:
TS inherits types here so you can write `setCount((prev) => prev + 1)`

Reply:
Yes, thank you